3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/zambia_focus_area_irs_base.psql

```diff
@@ -0,0 +1,212 @@
+			    MAX(latest_sa_event_date) AS latest_sa_event_date
+            FROM zambia_focus_area_irs AS zambia_focus_area_irs
+            WHERE zambia_focus_area_irs.plan_id = plans.identifier
+                AND ( (zambia_jurisdictions.is_virtual_jurisdiction
+                       AND zambia_focus_area_irs.is_virtual_jurisdiction
+                       AND zambia_focus_area_irs.jurisdiction_path @> ARRAY[zambia_jurisdictions.jurisdiction_parent_id])
+                     OR ((NOT zambia_jurisdictions.is_virtual_jurisdiction)
+                         AND (NOT zambia_focus_area_irs.is_virtual_jurisdiction)
+                         AND zambia_focus_area_irs.jurisdiction_path @> ARRAY[zambia_jurisdictions.jurisdiction_id]) )
+        ) AS zambia_focus_area_irs_query ON true
+        WHERE
+        -- dont get any spray areas
+        zambia_jurisdictions.is_leaf_node = FALSE
+        -- get only jurisdictions with spray areas
+        AND zambia_focus_area_irs_query.totAreas > 0
+    ) AS jurisdictions_query ON true
+    WHERE plans.intervention_type IN ('IRS', 'Dynamic-IRS') AND plans.status NOT IN ('draft', 'retired')
+) AS main_query
+ORDER BY CASE WHEN main_query.is_virtual_jurisdiction THEN 1 ELSE 0 END ASC, main_query.jurisdiction_name;
+
+CREATE INDEX IF NOT EXISTS zambia_irs_jurisdictions_path_idx_gin on pending_:schema .zambia_irs_jurisdictions using GIN(jurisdiction_path);
+CREATE INDEX IF NOT EXISTS zambia_irs_jurisdictions_plan_idx ON pending_:schema .zambia_irs_jurisdictions (plan_id);
+CREATE INDEX IF NOT EXISTS zambia_irs_jurisdictions_jurisdiction_idx ON pending_:schema .zambia_irs_jurisdictions (jurisdiction_id);
+CREATE INDEX IF NOT EXISTS zambia_irs_jurisdictions_jurisdiction_parent_idx ON pending_:schema .zambia_irs_jurisdictions (jurisdiction_parent_id);
+CREATE UNIQUE INDEX IF NOT EXISTS zambia_irs_jurisdictions_idx ON pending_:schema .zambia_irs_jurisdictions (id);
+
+-- Rename zambia_irs_jurisdictions in deprecated_
+ALTER MATERIALIZED VIEW deprecated_:schema .zambia_irs_jurisdictions RENAME TO zambia_irs_jurisdictions_base;
+
+ALTER INDEX IF EXISTS deprecated_:schema .zambia_irs_jurisdictions_path_idx_gin RENAME TO zambia_irs_jurisdictions_base_path_idx_gin;
+ALTER INDEX IF EXISTS deprecated_:schema .zambia_irs_jurisdictions_plan_idx RENAME TO zambia_irs_jurisdictions_base_plan_idx;
+ALTER INDEX IF EXISTS deprecated_:schema .zambia_irs_jurisdictions_jurisdiction_idx RENAME TO zambia_irs_jurisdictions_base_jurisdiction_idx;
+ALTER INDEX IF EXISTS deprecated_:schema .zambia_irs_jurisdictions_jurisdiction_parent_idx RENAME TO zambia_irs_jurisdictions_base_jurisdiction_parent_idx;
+ALTER INDEX IF EXISTS deprecated_:schema .zambia_irs_jurisdictions_idx RENAME TO zambia_irs_jurisdictions_base_idx;
+
+-- Rename current zambia_focus_area_irs view
+ALTER MATERIALIZED VIEW :"schema" .zambia_focus_area_irs RENAME TO zambia_focus_area_irs_base;
```

Comment:
Isn't this supposed to be moved to the top of this file so that it can be used in the queries above?  Otherwise what is the point of having this?

Reply:
updated